backend/src/services/analytics/anomaly_detection.py

```
@@ -1,38 +1,707 @@
+        if method == "zscore":
+            anomalies = self.detect_zscore_anomalies(
+                df['value'],
+                sensitivity=sensitivity
+            )
+        elif method == "isolation_forest":
+            anomalies = self.detect_isolation_forest_anomalies(
+                df[['value', 'count']],
+                contamination=0.05
+            )
+        else:
+            raise ValueError(f"Unknown detection method: {method}")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)